build_tools/benchmarks/collect_compilation_statistics.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env python3
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""Collect compilation statistics from benchmark suites.
+
+The benchmark suites need to be built with ninja and enable the CMake option
+IREE_ENABLE_COMPILATION_BENCHMARKS.
+"""
+
+import argparse
+from dataclasses import dataclass
+import json
+import os
+import re
+import zipfile
+from typing import Dict, Optional
+from pathlib import PurePath
+
+from common.benchmark_definition import CompilationInfo, CompilationResults, CompilationStatistics, ModuleComponentSizes, get_git_commit_hash
+from common.benchmark_suite import BENCHMARK_SUITE_REL_PATH, BenchmarkSuite
+
+BENCHMARK_FLAGFILE = "flagfile"
+MODULE_DIR = "vmfb"
+MODULE_FILE_EXTENSION = ".vmfb"
+NINJA_LOG_HEADER = "ninja log v5"
+NINJA_BUILD_LOG = ".ninja_log"
+COMPILATION_STATS_MODULE_SUFFIX = "compile-stats"
+
+VM_COMPONENT_NAME = "module.fb"
+CONST_COMPONENT_NAME = "_const.bin"
+DISPATCH_COMPONENT_PATTERNS = [
+    r".+_embedded_elf_.+\.so", r".+_vulkan_spirv_fb\.fb",
+    r".+_vmvx_bytecode_fb\.bin"
+]
```

Comment:
Nit: One per line or all on one line. This is a weird mix


```suggestion
DISPATCH_COMPONENT_PATTERNS = [
    r".+_embedded_elf_.+\.so",
    r".+_vulkan_spirv_fb\.fb",
    r".+_vmvx_bytecode_fb\.bin",
]
```